### PR TITLE
Use unique file names in tests

### PR DESCRIPTION
**Pull request by olpipi**
### Details:
 - Add api to common test utils to create unique names
 - Use this api in tests

### Tickets:
 - CVS-96479

### CODE REVIEW

**Review by ilyachur — COMMENTED** [review]
Can we grep and fix all test cases which works with non-unique file?

**Review by p-durandin — APPROVED** [review]
Please fix code style

**Review by ilyachur — COMMENTED** [review]
Please double check.
For example core tests...
I found 
 - https://github.com/openvinotoolkit/openvino/blob/master/src/core/tests/pass/serialization/const_compression.cpp#L16
 - https://github.com/openvinotoolkit/openvino/blob/master/src/core/tests/pass/serialization/tensor_iterator.cpp#L17
 
for example.

**Review by ilyachur — APPROVED** [review]

File: src/tests/functional/plugin/conformance/subgraphs_dumper/src/ops_cache.cpp [hunk 1]
```diff
@@ -196,10 +196,10 @@ OPCache::serialize_function(const std::pair<std::shared_ptr<ov::Node>, LayerTest
         }
         std::replace(op_name.begin(), op_name.end(), '/', '_');
         std::replace(op_name.begin(), op_name.end(), '\\', '_');
-        // TODO: Possible names collision
-        auto xml_path = current_op_folder + CommonTestUtils::FileSeparator + op_name + ".xml";
-        auto bin_path = current_op_folder + CommonTestUtils::FileSeparator + op_name + ".bin";
-        auto meta_info = current_op_folder + CommonTestUtils::FileSeparator + op_name + ".meta";
+        auto filePrefix = CommonTestUtils::generateTestFilePrefix();
```
Comment (iefode) [inline src/tests/functional/plugin/conformance/subgraphs_dumper/src/ops_cache.cpp:199]: @olpipi Why do you made such changes??? It causes of the crashes in `subgraphdumper`.
Next time ask me than make such changes
FYI @ilyachur

**Pull request merged**